src/aws_lambda_python/mpic_coordinator/domain/mpic_request.py

```diff
@@ -0,0 +1,46 @@
+from abc import ABC
+from typing import Literal
+
+from aws_lambda_python.common_domain.enum.check_type import CheckType
+from pydantic import BaseModel, model_validator
+
+from aws_lambda_python.mpic_coordinator.domain.mpic_orchestration_parameters import MpicRequestOrchestrationParameters
+from aws_lambda_python.common_domain.check_parameters import CaaCheckParameters, DcvCheckParameters
+from aws_lambda_python.common_domain.enum.dcv_validation_method import DcvValidationMethod
+
+
+class BaseMpicRequest(BaseModel, ABC):
```

Comment:
Yes I realized I was doing something really amateurish initially when I was just manually validating the structure of a `dict` instead of seeing what's out there to do this in a turn-key way. There's also FastAPI out there, which I might spend some cycles exploring to see if it's a good fit.

Reply:
I think its also worth noting that there are several tools to develop automatic validation from an Open API spec (e.g., https://www.npmjs.com/package/openapi-request-validator ) (AWS API Gateway can even be programmed using an Open API spec but I wanted to keep the spec a bit more platform neutral). An alternative approach to this validation system would be to write the spec as rich as possible and then just employ an Open API auto validator. I will note that there is a limit on expressiveness in Open API specs so this may have to co-exist with some manual validation.

Reply:
We could also use something like this as a testing tool: https://github.com/KissPeter/APIFuzzer

There are several options in the open api ecosystem if we choose to go that route:
https://tools.openapis.org/categories/data-validators.html

Although we should keep in mind which tools are available for python implementation.